### PR TITLE
Fix policy error on cascade archive

### PR DESCRIPTION
**Pull request by jechol**
### Contributor checklist

- [X] Bug fixes include regression tests
- [ ] Features include unit/acceptance tests

NOT FOR MERGE.

Just to reproduce policy error on cascading archive.

### CODE REVIEW

File: lib/ash_archival/resource/changes/archive_related.ex [hunk 1]
```diff
@@ -3,15 +3,16 @@ defmodule AshArchival.Resource.Changes.ArchiveRelated do
...
+  def change(changeset, _, context) do
     archive_related = AshArchival.Resource.Info.archive_related(changeset.resource)
 
     if Enum.empty?(archive_related) do
       changeset
     else
       Ash.Changeset.after_action(changeset, fn changeset, result ->
         # This is not optimized. We should do this with bulk queries, not resource actions.
-        loaded = changeset.api.load!(result, archive_related)
+        opts = context |> Keyword.new()
```
Comment (zachdaniel) [inline lib/ash_archival/resource/changes/archive_related.ex:14]: We can use `Ash.context_to_opts/1` to get `opts` here. This makes sense to do, but you've annotated this as not ready for merge. Does this change not resolve the issue?

**Comment by jechol** [comment]
@zachdaniel 
The issue is that archive_related maintains actor and tenant for children but not for grandchildren, causing the policy to fail. However, this seems to be a problem originating from ash rather than ash_archival.

The fact that actor is properly maintained for children can be confirmed by running mix test test/archival_with_policy_test.exs:209, while the fact that actor is not maintained for grandchildren can be confirmed by running mix test test/archival_with_policy_test.exs:227.

**Comment by zachdaniel** [comment]
Hmm…but ash core doesn’t really have a concept of “grandchildren”. It is just repeated calls to the change that should be passing the actor down. I’ll take a look tomorrow when I get the chance.

**Comment by jechol** [comment]
"Grandchildren" is a metaphorical term I used ;-)

**Comment by zachdaniel** [comment]
Your fix seems to do what you want, but I'm not convinced there is still an issue beyond that. Based on what I'm seeing, the failing test in your branch is because there is no policy that allows a non-admin user to destroy the record.

**Comment by zachdaniel** [comment]
LMK if you can confirm what I'm saying and we can merge this fix 👍

**Comment by jechol** [comment]
@zachdaniel I made mistake in test and fixed it now. Let's merge.

**Pull request merged**